CAPI Agent code changes for WILC platform.
Fixed compilation errors and warnings with buildroot 2017.08 and linaro toolchain

Signed-off-by: AdhamAbozaeid <adham.abozaeid@microchip.com>
---

diff -ruN Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/ca/wfa_ca.c Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/ca/wfa_ca.c
--- Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/ca/wfa_ca.c	2018-02-05 15:29:26.050604945 -0700
+++ Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/ca/wfa_ca.c	2018-02-05 16:07:42.455209000 -0700
@@ -336,6 +336,7 @@
 
             if(gSock > 0 && FD_ISSET(gSock, &sockSet))
                 {
+                	wfaTLV *tlvHdr;
                     DPRINT_INFO(WFA_OUT, "received from DUT\n");
                     sleep(1);
                     memset(respStr, 0, WFA_BUFF_128);
@@ -351,7 +352,8 @@
                         printf("%x ", caCmdBuf[i]);
                     printf("\n");
 #endif
-                    tag = ((wfaTLV *)caCmdBuf)->tag;
+                    tlvHdr = (wfaTLV *)caCmdBuf;
+                    tag = tlvHdr->tag;
 
                     memcpy(&ret_status, caCmdBuf+4, 4);
 
diff -ruN Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/console_src/wfa_con.c Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/console_src/wfa_con.c
--- Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/console_src/wfa_con.c	2018-02-05 15:29:26.050604945 -0700
+++ Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/console_src/wfa_con.c	2018-02-05 16:23:16.711814400 -0700
@@ -28,7 +28,12 @@
 #include <errno.h>
 #include <string.h>
 #include <pthread.h>
+ #include <unistd.h>
+ #include <arpa/inet.h>
 #include "wfa_con.h"
+#include "mpx.h"
+#include "../inc/wfa_utils.h"
+
 unsigned int rmsg[512];         // rx msg buffer
 unsigned int txmsg[512];        // tx msg buffer
 
@@ -248,7 +253,8 @@
         }
     }
 }
-main(int argc, char **argv)
+
+void main(int argc, char **argv)
 {
     int r, flags=0, n, i, id,base=10, bKeepFrom = 0;
     struct apts_msg *testcase;
diff -ruN Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/console_src/wfa_con.h Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/console_src/wfa_con.h
--- Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/console_src/wfa_con.h	2018-02-05 15:29:26.050604945 -0700
+++ Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/console_src/wfa_con.h	2018-02-05 16:27:44.902203500 -0700
@@ -104,3 +104,5 @@
     int myid;
     consoleProcStatetbl_t  *statefunc;
 };
+
+void WfaConResetAll();
\ No newline at end of file
diff -ruN Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/console_src/wfa_sndrcv.c Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/console_src/wfa_sndrcv.c
--- Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/console_src/wfa_sndrcv.c	2018-02-05 15:29:26.050604945 -0700
+++ Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/console_src/wfa_sndrcv.c	2018-02-05 16:31:33.138909200 -0700
@@ -20,6 +20,10 @@
 #include <netinet/in.h>
 #include <stdio.h>
 #include <string.h>
+ #include <unistd.h>
+#include "mpx.h"
+#include "../inc/wfa_utils.h"
+
 extern struct sockaddr dst;         // sock declarations
 extern unsigned int rmsg[512];      // rx msg buffer
 extern unsigned int txmsg[512];	    // tx msg buffer
diff -ruN Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/console_src/wfa_util.c Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/console_src/wfa_util.c
--- Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/console_src/wfa_util.c	2018-02-05 15:29:26.050604945 -0700
+++ Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/console_src/wfa_util.c	2018-02-05 16:38:10.078347500 -0700
@@ -23,7 +23,11 @@
 #include <net/if.h>
 #include <netinet/in.h>
 #include <unistd.h>
+#include <stdlib.h>
+ #include <arpa/inet.h>
 #include "wfa_con.h"
+#include "../inc/wfa_utils.h"
+
 extern struct sockaddr_in target;
 extern struct sockaddr_in from;
 extern struct sockaddr_in local;
@@ -36,6 +40,10 @@
 extern char traceflag;				// enable debug packet tracing
 int tos_vo,tos_vi,tos_be,tos_bk;
 void exit(int);
+int strvec_sep(char *s, char *array[], int n, char *sep);
+void pbaddr(char *s, unsigned long int in);
+int is_ipdotformat(char *s);
+
 char getipclass(unsigned long ip)
 {
 
@@ -444,7 +452,7 @@
     }
     return(id);
 }
-is_ipdotformat(char *s)
+int is_ipdotformat(char *s)
 {
     int d;
 
@@ -481,7 +489,7 @@
     return(i);
 }
 
-pbaddr(s, in)
+void pbaddr(s, in)
 char *s;
 unsigned long int in;
 {
diff -ruN Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/inc/wfa_miscs.h Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/inc/wfa_miscs.h
--- Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/inc/wfa_miscs.h	2018-02-05 15:29:26.054604945 -0700
+++ Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/inc/wfa_miscs.h	2018-02-05 15:00:05.450645584 -0700
@@ -23,11 +23,51 @@
 extern int isString(char *);
 extern int isNumber(char *);
 extern int isIpV4Addr(char *);
-extern inline double wfa_timeval2double(struct timeval *tval);
-extern inline void wfa_double2timeval(struct timeval *tval, double dval);
-extern inline double wfa_ftime_diff(struct timeval *t1, struct timeval *t2);
+inline double wfa_timeval2double(struct timeval *tval)
+{
+    return ((double) tval->tv_sec + (double) tval->tv_usec*1e-6);
+}
+
+inline void wfa_double2timeval(struct timeval *tval, double dval)
+{
+    tval->tv_sec = (long int) dval;
+    tval->tv_usec = (long int) ((dval - tval->tv_sec) * 1000000);
+}
+
+inline double wfa_ftime_diff(struct timeval *t1, struct timeval *t2)
+{
+    double dtime;
+    dtime = wfa_timeval2double(t2) - wfa_timeval2double(t1);
+    return dtime ;
+}
+
 extern int wfa_itime_diff(struct timeval *t1, struct timeval *t2);
-extern inline void int2BuffBigEndian(int val, char *buf);
-extern inline int bigEndianBuff2Int(char *buff);
+
+/*
+* The following two functions are converting Little Endian to Big Endian.
+* If your machine is already a Big Endian, you may flag it out.
+*/
+inline void int2BuffBigEndian(int val, char *buf)
+{
+    char *littleEn = (char *)&val;
+
+    buf[0] = littleEn[3];
+    buf[1] = littleEn[2];
+    buf[2] = littleEn[1];
+    buf[3] = littleEn[0];
+}
+
+inline int bigEndianBuff2Int(char *buff)
+{
+    int val;
+    char *strval = (char *)&val;
+
+    strval[0] = buff[3];
+    strval[1] = buff[2];
+    strval[2] = buff[1];
+    strval[3] = buff[0];
+
+    return val;
+}
 
 #endif
diff -ruN Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/inc/wfa_utils.h Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/inc/wfa_utils.h
--- Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/inc/wfa_utils.h	2018-02-05 15:29:26.054604945 -0700
+++ Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/inc/wfa_utils.h	2018-02-05 16:30:23.349124500 -0700
@@ -23,4 +23,14 @@
 
 #define WFA_RET_ENV       "WFA_UTIL_RET"
 
+int set_dscp(int new_dscp);
+int set_dscp0(int new_dscp);
+void create_apts_msg(int msg, unsigned int txbuf[],int id);
+void setup_socket();
+struct apts_msg * apts_lookup(char *s);
+int get_sta_id(unsigned int addr);
+int assign_sta_id(unsigned int addr);
+int expectedmsgrcd(unsigned int *rmsg,unsigned long type,int tos);
+int expectedmsgrcdl1(unsigned int *rmsg,unsigned long type,int tos);
+
 #endif
diff -ruN Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/lib/wfa_cmdproc.c Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/lib/wfa_cmdproc.c
--- Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/lib/wfa_cmdproc.c	2018-02-05 15:29:26.054604945 -0700
+++ Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/lib/wfa_cmdproc.c	2018-02-05 15:00:05.450645584 -0700
@@ -96,7 +96,7 @@
     if(strcasecmp(pcmdStr, pParam) == 0)
     {
         str = strtok_r(NULL, ",", &pcmdStr);
-        paramValue = atoi(str);
+        *paramValue = atoi(str);
         return 0;
     }
     return -1;
@@ -6842,26 +6842,26 @@
 		  else if (strcasecmp(str, "ADDED_SESSION") == 0)
 		  	staInvokeCmd->InvokeCmds.Msg.opcode= eMsgAddedSession;		  
 
-			for(;;)
-			{
+		for(;;)
+		{
+		  str = strtok_r(NULL, ",", &pcmdStr);
+		  if(str == NULL || str[0] == '\0')
+			 break;
+		  if (strcasecmp(str, "session_id") == 0)
+		  { 	   
+			 str = strtok_r(NULL, ",", &pcmdStr);
+			 staInvokeCmd->InvokeCmds.Msg.sessionId_flag= 1;
+			 staInvokeCmd->InvokeCmds.Msg.sessionID = strtol(str,NULL,16);				 
+		  }
+		  else if (strcasecmp(str, "session_mac") == 0) // not  defined in CAPI, just for information
+		  {
 			  str = strtok_r(NULL, ",", &pcmdStr);
-			  if(str == NULL || str[0] == '\0')
-				 break;
-			  if (strcasecmp(str, "session_id") == 0)
-			  { 	   
-				 str = strtok_r(NULL, ",", &pcmdStr);
-				 staInvokeCmd->InvokeCmds.Msg.sessionId_flag= 1;
-				 staInvokeCmd->InvokeCmds.Msg.sessionID = strtol(str,NULL,16);				 
-			  }
-			  else if (strcasecmp(str, "session_mac") == 0) // not  defined in CAPI, just for information
-			  {
-				  str = strtok_r(NULL, ",", &pcmdStr);
-				  staInvokeCmd->InvokeCmds.Msg.sessionMac_flag= 1;
-				  strncpy(staInvokeCmd->InvokeCmds.Msg.sessionMac,str,WFA_MAC_ADDR_STR_LEN-1);
-				  staInvokeCmd->InvokeCmds.Msg.sessionMac[WFA_MAC_ADDR_STR_LEN-1]='\0';
-			  }
+			  staInvokeCmd->InvokeCmds.Msg.sessionMac_flag= 1;
+			  strncpy(staInvokeCmd->InvokeCmds.Msg.sessionMac,str,WFA_MAC_ADDR_STR_LEN-1);
+			  staInvokeCmd->InvokeCmds.Msg.sessionMac[WFA_MAC_ADDR_STR_LEN-1]='\0';
+		  }
 
-			}
+		}
 
 		  
 	   }
@@ -7085,7 +7085,7 @@
 						 if (subtoken == NULL)
 							 break;
 						 strncpy(staManageServCmd->MngCmds.MgtServ.fileList[index],str,16);
-						 staManageServCmd->MngCmds.MgtServ.fileList[index][16]='\0';
+						 staManageServCmd->MngCmds.MgtServ.fileList[index][15]='\0';
 						 index++;
 					}
 					staManageServCmd->MngCmds.MgtServ.numModFiles= index;					
@@ -7103,7 +7103,7 @@
 						 if (subtoken == NULL)
 							 break;
 						 strncpy(staManageServCmd->MngCmds.MgtServ.modFileList[index],str,16);
-						 staManageServCmd->MngCmds.MgtServ.modFileList[index][16]='\0';
+						 staManageServCmd->MngCmds.MgtServ.modFileList[index][15]='\0';
 						 index++;
 					}
 					staManageServCmd->MngCmds.MgtServ.numModFiles= index;					
diff -ruN Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/lib/wfa_cs.c Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/lib/wfa_cs.c
--- Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/lib/wfa_cs.c	2018-02-05 15:29:26.054604945 -0700
+++ Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/lib/wfa_cs.c	2018-02-05 15:00:05.446645584 -0700
@@ -4266,7 +4266,7 @@
 {
 
 	dutCmdResponse_t infoResp;
-	caStaGetEventDetails_t *getStaGetEventDetails = (caStaMngServ_t *)caCmdBuf;  //uncomment and use it
+	caStaGetEventDetails_t *getStaGetEventDetails = (caStaGetEventDetails_t *)caCmdBuf;  //uncomment and use it
 	
 	 DPRINT_INFO(WFA_OUT, "\n Entry wfaStaGetEventDetails... ");
 
diff -ruN Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/lib/wfa_miscs.c Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/lib/wfa_miscs.c
--- Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/lib/wfa_miscs.c	2018-02-05 15:29:26.054604945 -0700
+++ Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/lib/wfa_miscs.c	2018-02-05 15:00:05.446645584 -0700
@@ -99,25 +99,6 @@
         return WFA_SUCCESS;
 }
 
-inline double wfa_timeval2double(struct timeval *tval)
-{
-    return ((double) tval->tv_sec + (double) tval->tv_usec*1e-6);
-}
-
-inline void wfa_double2timeval(struct timeval *tval, double dval)
-{
-    tval->tv_sec = (long int) dval;
-    tval->tv_usec = (long int) ((dval - tval->tv_sec) * 1000000);
-}
-
-inline double wfa_ftime_diff(struct timeval *t1, struct timeval *t2)
-{
-    double dtime;
-
-    dtime = wfa_timeval2double(t2) - wfa_timeval2double(t1);
-    return dtime ;
-}
-
 int wfa_itime_diff(struct timeval *t1, struct timeval *t2)
 {
     int dtime;
@@ -151,33 +132,6 @@
     return dtime;
 }
 
-/*
- * THe following two functions are converting Little Endian to Big Endian.
- * If your machine is already a Big Endian, you may flag it out.
- */
-inline void int2BuffBigEndian(int val, char *buf)
-{
-    char *littleEn = (char *)&val;
-
-    buf[0] = littleEn[3];
-    buf[1] = littleEn[2];
-    buf[2] = littleEn[1];
-    buf[3] = littleEn[0];
-}
-
-int bigEndianBuff2Int(char *buff)
-{
-    int val;
-    char *strval = (char *)&val;
-
-    strval[0] = buff[3];
-    strval[1] = buff[2];
-    strval[2] = buff[1];
-    strval[3] = buff[0];
-
-    return val;
-}
-
 int wfa_estimate_timer_latency()
 {
     struct timeval t1, t2, tp2;
diff -ruN Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/lib/wfa_sock.c Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/lib/wfa_sock.c
--- Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/lib/wfa_sock.c	2018-02-05 15:29:26.054604945 -0700
+++ Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/lib/wfa_sock.c	2018-02-05 15:00:05.446645584 -0700
@@ -333,7 +333,7 @@
 	
    sLen = wSTRLEN(ifname); 	
     wSTRNCPY(ifr.ifr_name, ifname, sLen);
-    ifr.ifr_name[sLen] = NULL;
+    ifr.ifr_name[sLen] = 0;
 
     ifr.ifr_addr.sa_family = AF_INET;
     if(wIOCTL(fd, SIOCGIFADDR, &ifr) == 0)
diff -ruN Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/lib/wfa_tg.c Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/lib/wfa_tg.c
--- Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/lib/wfa_tg.c	2018-02-05 15:29:26.054604945 -0700
+++ Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/lib/wfa_tg.c	2018-02-05 15:00:05.446645584 -0700
@@ -66,8 +66,7 @@
 
 extern double min_rttime;
 extern double gtgPktRTDelay;
-extern void int2BuffBigEndian(int val, char *buf);
-extern int bigEndianBuff2Int(char *buff);
+
 
 #ifdef WFA_WMM_PS_EXT
 extern int gtgWmmPS;
@@ -261,7 +260,6 @@
     int streamid = *(int *)(caCmdBuf);
     dutCmdResponse_t *stpResp = &gGenericResp;
     tgStream_t *myStream;
-    int i;
 
     stpResp->status = STATUS_COMPLETE;
 
@@ -1352,7 +1350,7 @@
     struct sockaddr_in    toAddr;
     char                  *packBuf=NULL; 
     int                   packLen, bytesSent, rate;
-    int                   sleepTimePerSent = 0, nOverTimeCount = 0, nDuration=0, nOverSend=0;
+    int                   nOverTimeCount = 0, nDuration=0, nOverSend=0;
     unsigned long long int sleepTotal=0, extraTimeSpendOnSending=0;   /* sleep mil-sec per sending  */
     int                   counter = 0, i, iSleep;     /*  frame data sending count */
     unsigned long         difftime;
@@ -1419,9 +1417,9 @@
     toAddr.sin_port = htons(theProf->dport); 
 
     /*  set sleep time per sending */
-    if ( theProf->rate < 100  )
-        sleepTimePerSent =  WFA_SEND_FIX_BITRATE_SLEEP_PER_SEND  +1; 
-    else sleepTimePerSent = WFA_SEND_FIX_BITRATE_SLEEP_PER_SEND;
+//    if ( theProf->rate < 100  )
+//        sleepTimePerSent =  WFA_SEND_FIX_BITRATE_SLEEP_PER_SEND  +1; 
+//    else sleepTimePerSent = WFA_SEND_FIX_BITRATE_SLEEP_PER_SEND;
 
     runLoop=1; /* global defined share with thread routine, should remove it later  */
     while ( runLoop)
@@ -1455,7 +1453,7 @@
                wfaSleepMilsec(1);
                
                sleepTotal = sleepTotal + (unsigned long long int) 1;
-               DPRINT_INFO(WFA_OUT, "wfaSendBitrateData wfaTrafficSendTo call ERR counter=%i i=%i; send busy, sleep %i MilSec then send\n", counter, i, sleepTimePerSent);
+               DPRINT_INFO(WFA_OUT, "wfaSendBitrateData wfaTrafficSendTo call ERR counter=%i i=%i; send busy\n", counter, i);
                i--;
            }
            /*  sleep per batch sending */
diff -ruN Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/lib/wfa_thr.c Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/lib/wfa_thr.c
--- Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0.orig/lib/wfa_thr.c	2018-02-05 15:29:26.054604945 -0700
+++ Wi-FiTestSuite_Sample_DUT_Code-Linux_v9.2.0/lib/wfa_thr.c	2018-02-05 15:00:05.446645584 -0700
@@ -437,7 +437,7 @@
 int WfaStaSndVO(char psave,int sleep_period,int *state)
 {
     int r;
-    static int en=1;
+
     PRINTF("\r\nEnterring WfastasndVO %d",en++);
     if ((r=sender(psave,sleep_period,TG_WMM_AC_VO))>=0)
         (*state)++;
@@ -455,7 +455,6 @@
 int WfaStaSnd2VO(char psave,int sleep_period,int *state)
 {
     int r;
-    static int en=1;
 
     PRINTF("\r\nEnterring WfastasndVO %d",en++);
     if ((r=sender(psave,sleep_period,TG_WMM_AC_VO))>=0)
@@ -478,7 +477,6 @@
 int WfaStaSndVI(char psave,int sleep_period,int *state)
 {
     int r;
-    static int en=1;
 
     PRINTF("\r\nEnterring WfastasndVI %d",en++);
     if ((r=sender(psave,sleep_period,TG_WMM_AC_VI))>=0)
@@ -495,7 +493,6 @@
 int WfaStaSndBE(char psave,int sleep_period,int *state)
 {
     int r;
-    static int en=1;
 
     PRINTF("\r\nEnterring WfastasndBE %d",en++);
     if ((r=sender(psave,sleep_period,TG_WMM_AC_BE))>=0)
@@ -511,7 +508,6 @@
 int WfaStaSndBK(char psave,int sleep_period,int *state)
 {
     int r;
-    static int en=1;
 
     PRINTF("\r\nEnterring WfastasndBK %d",en++);
     if ((r=sender(psave,sleep_period,TG_WMM_AC_BK))>=0)
@@ -528,7 +524,6 @@
 int WfaStaSndVOCyclic(char psave,int sleep_period,int *state)
 {
     int i;
-    static int en=1;
 
     for(i=0; i<3000; i++)
     {
@@ -596,7 +591,7 @@
     tgWMM_t *my_wmm = &wmm_thr[myId];
     tgStream_t *myStream = NULL;
     int myStreamId, i=0,rttime=0,difftime=0, rcvCount=0,sendCount=0;
-    int mySock = -1, status, respLen = 0, nbytes = 0, ret=0, j=0;
+    int mySock = -1, status, respLen = 0, nbytes = 0, j=0;
     tgProfile_t *myProfile;
     pthread_attr_t tattr;
 #ifdef WFA_WMM_PS_EXT
@@ -604,10 +599,10 @@
     StationProcStatetbl_t  curr_state;
 #endif
 
-//#ifdef WFA_VOICE_EXT
     struct timeval lstime, lrtime;
+#ifdef WFA_VOICE_EXT
     int asn = 1;  /* everytime it starts from 1, and to ++ */
-//#endif
+#endif
 
     wPT_ATTR_INIT(&tattr);
     wPT_ATTR_SETSCH(&tattr, SCHED_RR);
@@ -742,7 +737,7 @@
 
                 tmout.tv_sec = 0;
                 tmout.tv_usec = 15000;     // set for 15000 microsec timeout for rcv              
-                ret = setsockopt(mySock, SOL_SOCKET, SO_RCVTIMEO, (char *)&tmout, (socklen_t) sizeof(tmout)); 
+                setsockopt(mySock, SOL_SOCKET, SO_RCVTIMEO, (char *)&tmout, (socklen_t) sizeof(tmout)); 
                 
                 rcvCount=0; sendFailCount=0;
                 j=0;  sendCount=0;
